api/v1alpha1/trafficmanager_types.go

```diff
@@ -0,0 +1,264 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type TrafficManagerProfile struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec TrafficManagerProfileSpec `json:"spec"`
+	// +optional
+	Status TrafficManagerProfileStatus `json:"status,omitempty"`
+}
+
+type TrafficManagerProfileSpec struct {
+	// Type of routing method.
+	// +kubebuilder:validation:Enum=Weighted
+	// +kubebuilder:default=Weighted
+	// +optional
+	RoutingMethod TrafficRoutingMethod
+
+	// The DNS settings of the Traffic Manager profile.
+	// +required
+	DNSConfig *DNSConfig
+
+	// The endpoint monitoring settings of the Traffic Manager profile.
+	MonitorConfig *MonitorConfig
+
+	// The list of endpoints in the Traffic Manager profile.
+	Endpoints []*TrafficManagerEndpointRef
+}
+
+type TrafficManagerEndpointRefType string
+
+const (
+	// TrafficManagerEndpointTypeServiceImport is a group of public ip exposed by the exported service.
+	TrafficManagerEndpointTypeServiceImport TrafficManagerEndpointRefType = "ServiceImport"
+)
+
+type TrafficManagerEndpointRef struct {
+	// Type of endpoint reference. Can be "ServiceImport". Default is ServiceImport.
+	// +kubebuilder:validation:Enum=ServiceImport
+	// +kubebuilder:default=ServiceImport
+	// +optional
+	Type TrafficManagerEndpointRefType
```

Comment:
What other types will we support? The type is defined in this level means one TM can support mixed end point types, is that the intention?